src/System.Net.Http/tests/FunctionalTests/ManagedHandlerTest.cs

```diff
@@ -261,7 +263,52 @@ public override int Read(byte[] buffer, int offset, int count)
             public override long Seek(long offset, SeekOrigin origin) => throw new NotImplementedException();
             public override void SetLength(long value) => throw new NotImplementedException();
         }
+    }
+
+    public sealed class ManagedHandler_ConnectionUpgrade_Test : HttpClientTestBase
+    {
+        protected override bool UseManagedHandler => true;
+
+        [Fact]
+        public async Task UpgradeConnection_Success()
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    // We need to use ResponseHeadersRead here, otherwise we will hang trying to buffer the response body.
+                    Task<HttpResponseMessage> getResponseTask = client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+                    await LoopbackServer.AcceptSocketAsync(server, async (s, stream, reader, writer) =>
+                    {
+                        Task<List<string>> serverTask = LoopbackServer.ReadWriteAcceptedAsync(s, reader, writer,
+                            $"HTTP/1.1 101 Switching Protocols\r\nDate: {DateTimeOffset.UtcNow:R}\r\n\r\n");
+
+                        await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
+
+                        using (Stream clientStream = await (await getResponseTask).Content.ReadAsStreamAsync())
+                        {
+                            Assert.True(clientStream.CanWrite);
+                            Assert.True(clientStream.CanRead);
+                            Assert.False(clientStream.CanSeek);
+
+                            TextReader clientReader = new StreamReader(clientStream);
+                            TextWriter clientWriter = new StreamWriter(clientStream) { AutoFlush = true };
+
+                            clientWriter.WriteLine("hello server");
```

Comment:
nit: all these literal strings which are duplicated across the following lines could be put into a some const string variables.

Reply:
Will do